src/Agent.Listener/Configuration/Validators.cs

```diff
@@ -92,5 +95,24 @@ public static bool NTAccountValidator(string arg)
 
             return true;
         }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "locationServer")]
+        public static async Task<bool> IsHostedServer(string serverUrl, VssCredentials credentials, ILocationServer locationServer)
```

Comment:
Could you please explain why this method was included in the Validators class in particular?

Reply:
Since this method validates whether a server is hosted or not, returns true or false, I suppose it makes sense to include it in the Validators class.

Reply:
Not sure if this is semantically good place to locate it - validators here are indented to verify different data types - although IsHostedServer is intended to retrieve the server type.

Reply:
Moved to the separate module